internal/hcsoci/resources_lcow.go

```diff
@@ -120,6 +120,19 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, r *
 				// Mounts that map to a path in UVM are specified with 'sandbox://' prefix.
 				// example: sandbox:///a/dirInUvm destination:/b/dirInContainer
 				uvmPathForFile = mount.Source
+			} else if strings.HasPrefix(mount.Source, "hugepages://") {
+				// currently we only support 2M hugepage size
+				hugePageSubDirs := strings.Split(strings.TrimPrefix(mount.Source, "hugepages://"), "/")
+				if len(hugePageSubDirs) < 2 {
+					return errors.Errorf(`hugepages://%s' mount path is invalid, expected format: hugepages://<hugepage-size>/<hugepage-src-location>`, mount.Source)
+				}
+
+				// hugepages:// should be followed by pagesize
+				if hugePageSubDirs[0] != "2M" {
+					return errors.Errorf(`Only 2M (megabytes) pagesize is supported.`)
```

Comment:
The `Errorf` function (and any of the family of functions that end with f generally) expect to receive a format string with values to get filled in e.g. `return errors.Errorf("only 2M pagesize is supported, got %s", hugePageSubDirs[0])` or something similar. We also just recently added a linter step to detect when errors start with capital letters and end with punctuation as that's the go idiom (https://github.com/golang/go/wiki/CodeReviewComments#error-strings) so best to follow this. I believe this isn't rebase against the most recent master here as I don't see the new linter step in the PR but that's fine.